/*
Copyright 2017-2018 Echo Park Labs

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

For additional information, contact:

email: info@echoparklabs.io
*/

//    https://github.com/protocolbuffers/protobuf/issues/1606
syntax = "proto2";

option java_multiple_files = true;
option java_package = "com.epl.protobuf.stac";
option java_outer_classname = "StacMetadata";
option objc_class_prefix = "STC";

package epl.protobuf;

import "google/protobuf/timestamp.proto";
import "epl/protobuf/geometry_operators.proto";
import "epl/protobuf/stac.proto";

/*
https://groups.google.com/forum/#!topic/protobuf/bO2L6-_t91Q
When mixing proto2 and proto3, there's a checklist of don't (like no 'required' fields)
1. All fields should be optional.
2. Do not use custom default values.
3. All enums should have first element with tag = 0.
4. Do not use extension for anything except custom options.
*/

message Asset {
    optional string href = 1;
    optional ASSET_TYPE asset_type = 2;
    optional EO_BAND eo_band = 3;
    optional IAAS bucket_iaas_host = 4;
    optional string bucket_owner = 5;
    optional string bucket_region = 6;
    optional string bucket_ref = 7;
}

message MetadataResult {
    optional string id = 1;
    optional EO_PLATFORM eo_platform = 2;
    optional EO_INSTRUMENT eo_instrument = 3;
    optional float eo_cloud_cover = 6;
    optional float eo_sun_azimuth = 7;
    optional float eo_sun_elevation = 8;
    optional float eo_gsd = 9;
    optional float eo_off_nadir = 10;

    optional epl.protobuf.EnvelopeData bbox = 11;
    optional epl.protobuf.GeometryBagData geometry = 12;
    map<string, Asset> assets = 13;

    reserved 14 to 200;

    optional google.protobuf.Timestamp src_img_date = 201;
    optional string usgsid = 202;
}


